monkey/infection_monkey/master/automated_master.py

```diff
@@ -225,8 +225,10 @@ def _run_plugins(
 
         interrupted_message = f"Received a stop signal, skipping remaining {plugin_type}s"
         for p in interruptible_iter(plugins, self._stop, interrupted_message):
-            # TODO: Catch exceptions to prevent thread from crashing
-            callback(p)
+            try:
+                callback(p)
+            except Exception as ex:
+                logger.debug(f"Exception encountered when running {plugin_type} {p}: {str(ex)}")
```

Comment:
Why do we want to catch all exceptions here? And most of all why we log uncaught exceptions as debug level?

Reply:
So the thread doesn't crash? Will change the log level to warning.